### PR TITLE
clean up field-reference-parser

### PR DESCRIPTION
**Pull request by kaisecheng**
Fixed: #12355

remove field-reference-parser for version 8

### CODE REVIEW

**Review by andsel — APPROVED** [review]
LGTM

File: docs/static/breaking-changes.asciidoc [hunk 1]
```diff
@@ -24,6 +25,21 @@ See also <<releasenotes>>.
 //NOTE: The notable-breaking-changes tagged regions are re-used in the
 //Installation and Upgrade Guide
 
+[[breaking-8.0]]
```
Comment (kaisecheng) [inline docs/static/breaking-changes.asciidoc:28]: @karenzone I should cc you before merging it :fearful:
Could you review the doc? This PR is a breaking change of removing `--field-reference-parser`

File: docs/static/breaking-changes.asciidoc [hunk 2]
```diff
@@ -24,6 +25,21 @@ See also <<releasenotes>>.
...
+[float]
+==== Changes in Logstash Core
+
+[float]
+[[field-reference-parser]]
+===== Field Reference parser removed
+The Field Reference parser by default was STRICT in version 7. This configuration is removed permanently.
+Logstash keeps the behaviour as if the default setting, which interprets references to fields in your pipelines and plugins,
+and rejects inputs that are either ambiguous or illegal.
+
```
Comment (karenzone) [inline docs/static/breaking-changes.asciidoc:41]: What about:
"The Field Reference parser interprets references to fields in your pipelines and
plugins. It was configurable in 7.x, with the default set to `strict` to reject
inputs that are ambiguous or illegal. Configurability is  removed in 8.0. Now
{ls] rejects ambiguous and illegal inputs as standard behavior."

**Review by karenzone — COMMENTED** [review]
I offered some minor wording changes in line for your consideration.  

What you have does not break the build, and will be acceptable for now if you choose not to make the changes now.

**Pull request merged**